internal/fans/nvidia.go

```diff
@@ -0,0 +1,228 @@
+package fans
+
+import (
+	"errors"
+	"os"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	"github.com/markusressel/fan2go/internal/configuration"
+	"github.com/markusressel/fan2go/internal/ui"
+)
+
+type NvidiaFan struct {
+	Label             string                  `json:"label"`
+	Index             int                     `json:"index"`
+	RpmMovingAvg      float64                 `json:"rpmMovingAvg"`
+	Config            configuration.FanConfig `json:"config"`
+	MinPwm            *int                    `json:"minPwm"`
+	StartPwm          *int                    `json:"startPwm"`
+	MaxPwm            *int                    `json:"maxPwm"`
+	FanCurveData      *map[int]float64        `json:"fanCurveData"`
+	Rpm               int                     `json:"rpm"`
+	Pwm               int                     `json:"pwm"`
+	RunAtMaxSpeed     bool                    // to emulate PWM mode 0
+	SupportedFeatures int                     // (1 << FeaturePwmSensor) | (1 << FeatureRpmSensor) or similar
+	// TODO: probably SupportedFeatures doesn't have to be saved to config? set in GetDevices()
+	// TODO: put nvml.Device here? though in reality it should be shared
+	//   between all fans and sensors of that device (but multiple devices can exist
+	//   when the system has multiple GPUs)
+}
+
+// helper function to turn an nvml error/return code into a go error
+// (also handles success by returning nil)
+func nvError(errCode nvml.Return) error {
+	if errCode == nvml.SUCCESS {
+		return nil
+	}
+	return errors.New(nvml.ErrorString(errCode))
+}
+
+func (fan *NvidiaFan) getNvFanIndex() int {
+	// seems like fan indices in fan2go are 1-based,
```

Comment:
correct, this is a result of the `fan2go detect` output, where I wanted to avoid starting with `0`.

Reply:
ok